Lib/fontTools/cffLib/CFF2ToCFF.py

```diff
@@ -0,0 +1,166 @@
+"""CFF to CFF2 converter."""
+
+from fontTools.ttLib import TTFont, newTable
+from fontTools.misc.cliTools import makeOutputFileName
+from fontTools.cffLib import TopDictIndex, buildOrder, topDictOperators
+from .width import optimizeWidths
+from collections import defaultdict
+import logging
+
+
+__all__ = ["convertCFF2ToCFF", "main"]
+
+
+log = logging.getLogger("fontTools.cffLib")
+
+
+def _convertCFF2ToCFF(cff, otFont):
+    """Converts this object from CFF2 format to CFF format. This conversion
+    is done 'in-place'. The conversion cannot be reversed.
+
+    The CFF2 font cannot be variable. (TODO Accept those and convert to the
```

Comment:
is the input CFF2 also assumed to _not_ be already subroutinized? Or it doesn't matter in this case?

Reply:
oh I see you do seem to handle GlobalSubrs and Private.Subrs in here. I was simply wondering if inserting the optimal width at index 0 can be done unconditionally even in presence of subroutines. Ignore me if it doesn't make any sense

Reply:
Yeah there's no problem with subroutines. I first compute optimal default and nominal widths from the hmtx table data, then just insert at index 0 the width for those glyphs that need it. It works with subroutines as well.